Real forward test fails

OS : Linux, 
Compiler:  GCC 9.3.0,
Arch: avx2 opt;
Version: 0.6.0

```
-------------------------------------------------------------------------------
testing autodiff::real
-------------------------------------------------------------------------------
/home/coulombc/autodiff/tests/forward/real/real.test.cpp:61
...............................................................................

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:356: FAILED:
  CHECK( y[0] == approx(abs(x[0])) )
with expansion:
  0.5 == Approx( 0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:357: FAILED:
  CHECK( y[1] == approx(abs(x[0])/x[0] * x[1]) )
with expansion:
  3.0 == Approx( 0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:358: FAILED:
  CHECK( y[2] == approx(abs(x[0])/x[0] * x[2]) )
with expansion:
  -5.0 == Approx( -0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:359: FAILED:
  CHECK( y[3] == approx(abs(x[0])/x[0] * x[3]) )
with expansion:
  -15.0 == Approx( -0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:360: FAILED:
  CHECK( y[4] == approx(abs(x[0])/x[0] * x[4]) )
with expansion:
  11.0 == Approx( 0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:365: FAILED:
  CHECK( z[0] == approx(abs(y[0])) )
with expansion:
  0.5 == Approx( 0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:366: FAILED:
  CHECK( z[1] == approx(abs(y[0])/(y[0]) * y[1]) )
with expansion:
  3.0 == Approx( 0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:367: FAILED:
  CHECK( z[2] == approx(abs(y[0])/(y[0]) * y[2]) )
with expansion:
  -5.0 == Approx( -0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:368: FAILED:
  CHECK( z[3] == approx(abs(y[0])/(y[0]) * y[3]) )
with expansion:
  -15.0 == Approx( -0.0 )

/home/coulombc/autodiff/tests/forward/real/real.test.cpp:369: FAILED:
  CHECK( z[4] == approx(abs(y[0])/(y[0]) * y[4]) )
with expansion:
  11.0 == Approx( 0.0 )

===============================================================================
test cases:   10 |    9 passed |  1 failed
assertions: 2523 | 2513 passed | 10 failed

```

One thing I've noticed building debug with Clang is a large number of warnings about using integer `abs` with floating point values, e.g.:
```
./tests/forward/real/real.test.cpp:358:25: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
    CHECK_APPROX( y[2], abs(x[0])/x[0] * x[2] );
```
That said, this is reported for files with tests that pass, so I doubt it's the cause of the test failures.

--- a/tests/forward/dual/dual.test.cpp
+++ b/tests/forward/dual/dual.test.cpp
@@ -451,10 +451,10 @@ TEST_CASE("testing autodiff::dual", "[forward][dual]")
         CHECK_DERIVATIVES_FX(pow(x, 2.0 * x), pow(val(x), 2.0 * val(x)), 2.0 * (log(x) + 1) * pow(x, 2.0 * x));
 
         // Testing abs function (when x > 0)
-        x = 1.0; CHECK_DERIVATIVES_FX(abs(x), abs(val(x)), 1.0);
+        x = 1.0; CHECK_DERIVATIVES_FX(abs(x), std::abs(val(x)), 1.0);
 
         // Testing abs function (when x < 0)
-        x = -1.0; CHECK_DERIVATIVES_FX(abs(x), abs(val(x)), -1.0);
+        x = -1.0; CHECK_DERIVATIVES_FX(abs(x), std::abs(val(x)), -1.0);
 
         // Testing erf function (when x = 1.0)
         x =  1.0; CHECK_DERIVATIVES_FX(erf(x), erf(val(x)), 0.4151074974);

--- a/tests/forward/real/real.test.cpp
+++ b/tests/forward/real/real.test.cpp
@@ -353,20 +353,20 @@ TEST_CASE("testing autodiff::real", "[forward][real]")
 
     y = abs(x);
 
-    CHECK_APPROX( y[0], abs(x[0]) );
-    CHECK_APPROX( y[1], abs(x[0])/x[0] * x[1] );
-    CHECK_APPROX( y[2], abs(x[0])/x[0] * x[2] );
-    CHECK_APPROX( y[3], abs(x[0])/x[0] * x[3] );
-    CHECK_APPROX( y[4], abs(x[0])/x[0] * x[4] );
+    CHECK_APPROX( y[0], std::abs(x[0]) );
+    CHECK_APPROX( y[1], std::abs(x[0])/x[0] * x[1] );
+    CHECK_APPROX( y[2], std::abs(x[0])/x[0] * x[2] );
+    CHECK_APPROX( y[3], std::abs(x[0])/x[0] * x[3] );
+    CHECK_APPROX( y[4], std::abs(x[0])/x[0] * x[4] );
 
     y = -x;
     z = abs(y);
 
-    CHECK_APPROX( z[0], abs(y[0]) );
-    CHECK_APPROX( z[1], abs(y[0])/(y[0]) * y[1] );
-    CHECK_APPROX( z[2], abs(y[0])/(y[0]) * y[2] );
-    CHECK_APPROX( z[3], abs(y[0])/(y[0]) * y[3] );
-    CHECK_APPROX( z[4], abs(y[0])/(y[0]) * y[4] );
+    CHECK_APPROX( z[0], std::abs(y[0]) );
+    CHECK_APPROX( z[1], std::abs(y[0])/(y[0]) * y[1] );
+    CHECK_APPROX( z[2], std::abs(y[0])/(y[0]) * y[2] );
+    CHECK_APPROX( z[3], std::abs(y[0])/(y[0]) * y[3] );
+    CHECK_APPROX( z[4], std::abs(y[0])/(y[0]) * y[4] );
 
     //=====================================================================================================================
     //

